### ISSUE TITLE
coreos's etcd container images don't specify appc.io/executor/supports-systemd-notify

### ISSUE THREAD

**Issue by multi-io** [coreos/etcd#7840]
The coreos.com/etcd images do not contain the `appc.io/executor/supports-systemd-notify` annotation even though etcd does support `sd_notify`. 

I'm not sure about the exact effects that this has; it would certainly depend on the container runtime -- with rkt, if you run in a coreos stage1, i.e. with proper namespacing and an intermediary in-container systemd that runs etcd in a dynamically created service unit, it seems that rkt creates that service unit with type=simple rather than type=notify (which would be used if the image declared the annotation properly), which leads to the inner systemd signaling the etcd service readiness "too early", i.e. immediately after the etcd process has been started rather than after it has send sd_notify.

I don't know how the coreos.com/etcd image is built, so I'm not sure how to fix this. If the `scripts/build-aci` script is used for this, it would be a one-line fix I suppose.

**Comment by gyuho** [comment]
Seems like `rkt` added `sd_notify`support since `1.15.0`? https://github.com/rkt/rkt/blob/master/CHANGELOG.md#1150

Isn't this `rkt` runtime issue?

**Comment by philips** [comment]
appc is deprecated and we shouldn't be adding new features or fixing appc issues: https://github.com/appc/spec#-disclaimer-

**Comment by multi-io** [comment]
@philips Yeah appc is deprecated but that annotation is not, afaict. It's still referenced in the docs: https://coreos.com/rkt/docs/latest/using-rkt-with-systemd.html#managing-pods-as-systemd-services . So if it is deprecated, there's a documentation issue :)

**Comment by multi-io** [comment]
@gyuho I may be wrong, but it seems to be rkt and the container image (etcd in this case) working in concert, as I tried to explain in the ticket: Rkt (stage1-coreos.aci) checks whether the image has that annotation, and if it does, the in-container systemd waits for sd_notify from the container process (which it has wrapped in a dynamically created type=notify service) and when it received it, it passes it on to the host systemd. If the annotation is missing from the image, the container process is wrapped in a type=simple service instead and the host systemd is sd_notified as soon as the container service has started, i.e. potentially before it has sent its sd_notify. And that might be too early.

At least that's what I inferred from skimming the rkt source for half an hour or so.
